### PR TITLE
[Doc] Set `action_spec` instead of `input_spec`

### PR DESCRIPTION
**Pull request by FrankTianTT**
## Description

`input_spec` is protected and does not support with direct setting. Example code in `CEM` and `MPPI` documentation will cause error:
```
Traceback (most recent call last):
  File "/Users/frank/Projects/target-domain-fast-adaptation/tdfa/mpc/try.py", line 50, in <module>
    env = MyMBEnv(world_model)
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torchrl/envs/common.py", line 137, in __call__
    instance: EnvBase = super().__call__(*args, **kwargs)
  File "/Users/frank/Projects/target-domain-fast-adaptation/tdfa/mpc/try.py", line 19, in __init__
    self.input_spec = CompositeSpec(
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torchrl/envs/common.py", line 315, in __setattr__
    return super().__setattr__(key, value)
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1754, in __setattr__
    super().__setattr__(name, value)
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torchrl/envs/common.py", line 427, in input_spec
    raise RuntimeError("input_spec is protected.")
RuntimeError: input_spec is protected.
```
Let's set the `action_spec` and let the program decide the `input_spec`.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by vmoens — COMMENTED** [review]
Good catch! 
The remaining field in input spec should be placed in `state_spec` which can be edited.
```python
self.state_spec = CompositeSpec(...)
```

**Comment by FrankTianTT** [comment]
@vmoens I have corrected it, if we are on the same page:)

**Comment by FrankTianTT** [comment]
> I have corrected it, if we are on the same page:)

@vmoens Sorry, I just checked and directly changing `self.observation_spec` to `self.state_spec` will cause new problems:
```
Traceback (most recent call last):
  File "/Users/frank/Projects/target-domain-fast-adaptation/tdfa/mpc/try2.py", line 54, in <module>
    planner = MPPIPlanner(
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torchrl/modules/planners/mppi.py", line 136, in __init__
    super().__init__(env=env, action_key=action_key)
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torchrl/modules/planners/common.py", line 37, in __init__
    in_keys = list(env.observation_spec.keys(True, True))
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torchrl/envs/common.py", line 1159, in observation_spec
    observation_spec = self.output_spec["full_observation_spec"]
  File "/Users/frank/opt/anaconda3/envs/tdfa/lib/python3.8/site-packages/torchrl/data/tensor_specs.py", line 3271, in __getitem__
    return self._specs[idx_unravel]
KeyError: 'full_observation_spec'
```

This seems to be because `MPCPlannerBase` expects an `observation_spec` that can normalize the input to the algorithm. In the example code of `ModelBasedEnvBase`, `MyMBEnv` initializes both `observation_spec` and `state_spec`:
```python
...         self.observation_spec = CompositeSpec(
...             hidden_observation=UnboundedContinuousTensorSpec((4,))
...         )
...         self.state_spec = CompositeSpec(
...             hidden_observation=UnboundedContinuousTensorSpec((4,)),
...         )
```

Actually this caused me more doubts, are the `state_spec` and `observation_spec` above the same thing? What I mean is that `ModelBasedEnvBase` is a stateless environment, and its `step` should accept $s_t, a_t$ and return $s_{t+1}, r_{t+1}$. At this time, whether `state_spec` corresponds to $s_t$ and `observation_spec` corresponds to $s_{t+1}$?

More specifically, if I want to use `ModelBasedEnvBase` to learn an environment (a stateful environment, such as gym or dmc), should its `observation_spec` and `state_spec` both be set to the `observation_spec` of the environment being learned, In order to support both model learning and MPC at the same time?

File: torchrl/modules/planners/cem.py [hunk 1]
```diff
@@ -51,13 +51,10 @@ class CEMPlanner(MPCPlannerBase):
         >>> class MyMBEnv(ModelBasedEnvBase):
         ...     def __init__(self, world_model, device="cpu", dtype=None, batch_size=None):
         ...         super().__init__(world_model, device=device, dtype=dtype, batch_size=batch_size)
-        ...         self.observation_spec = CompositeSpec(
+        ...         self.state_spec = CompositeSpec(
```
Comment (vmoens) [inline torchrl/modules/planners/cem.py:54]: What I had in mind is
```suggestion
        ...         self.state_spec = CompositeSpec(
        ...             hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
        ...         self.observation_spec = CompositeSpec(
        ...             next_hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
```

and here's the full code

```python
        >>> class MyMBEnv(ModelBasedEnvBase):
        ...     def __init__(self, world_model, device="cpu", dtype=None, batch_size=None):
        ...         super().__init__(world_model, device=device, dtype=dtype, batch_size=batch_size)
        ...         self.state_spec = CompositeSpec(
        ...             hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
        ...         self.observation_spec = CompositeSpec(
        ...             hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
        ...         self.action_spec = UnboundedContinuousTensorSpec((1,))
        ...         self.reward_spec = UnboundedContinuousTensorSpec((1,))
        ...
        ...     def _reset(self, tensordict: TensorDict) -> TensorDict:
        ...         tensordict = TensorDict(
        ...             {},
        ...             batch_size=self.batch_size,
        ...             device=self.device,
        ...         )
        ...         tensordict = tensordict.update(
        ...             self.state_spec.rand())
        ...         tensordict = tensordict.update(
        ...             self.observation_spec.rand())
        ...         return tensordict
```

Not sure if the observation should be named differently from the input, IDK where that comes from.

I think that with this implementation, the input at `t+1` will match the output from `t` which satisfies the autoregressive property of the planner

Comment (FrankTianTT) [reply torchrl/modules/planners/cem.py:54]: I see. I will modify according to your ideas and ensure that there are no errors.

**Review by vmoens — APPROVED** [review]
Thanks for working on this!

**Pull request merged**